src/pages/index.js

```diff
@@ -73,57 +68,195 @@ function handleImageClick(cardData) {
+  ".cards__list"
 );
 
-cardSection.renderItems();
+//Event Listeners
 
-// Render Cards
-function renderCard(item, method = "addItem") {
-  const cardElement = createCard(item);
-  cardSection[method](cardElement);
-}
+addNewCardButton.addEventListener("click", () => {
+  addCardFormPopup.open();
+});
+
+profileEditButton.addEventListener("click", () => {
+  // editProfileFormValidator.resetValidation();
+  const inputData = userInfo.getUserInfo();
+  // console.log(inputData);
+  profileNameInput.value = inputData.name;
+  profileDescriptionInput.value = inputData.about;
+  profileEditPopup.open();
+});
+
+addCardFormValidator.enableValidation();
+editProfileFormValidator.enableValidation();
 
 //Functions
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
-  return card.getView();
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteCardSubmit,
+    likeCard,
+    unlikeCard
+  );
+  return card.generateCard();
 }
 
 function handleProfileEditSubmit(inputData) {
-  userInfo.setUserInfo({
-    name: inputData.name,
-    job: inputData.description,
-  });
+  profileEditPopup.renderLoading(true);
+
+  api
+    .setUserInfo(inputData.name, inputData.description)
+    .then(() => {
+      userInfo.setUserInfo({
+        name: inputData.name,
+        about: inputData.description,
+      });
+      profileEditPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error updating user info", err);
+      alert(err);
+    })
+    .finally(() => {
+      console.log("Edit form complete");
+      profileEditPopup.renderLoading(false);
+    });
 
-  profileEditPopup.close();
+  // profileEditPopup.close();
 }
 
 function handleAddCardSubmit(inputData) {
-  const cardData = {
-    name: inputData.title,
-    link: inputData.url,
-  };
-  renderCard(cardData);
-  addCardFormValidator.disableButton();
-  addCardFormPopup.close();
-  addCardForm.reset();
+  const name = inputData.title;
+  const link = inputData.url;
+
+  addCardFormPopup.renderLoading(true);
+  api
+    .uploadCard({ name, link })
+    .then((cardData) => {
+      const card = createCard(cardData);
+      // cardAddForm.setLoading(false);
+      cardSection.addItem(card);
+      addCardFormPopup.close();
+      addCardForm.reset();
+    })
+    .catch((error) => {
+      console.error(error);
+    })
+    .finally(() => {
+      console.log("Add card complete");
+      addCardFormPopup.renderLoading(false);
+    });
 }
 
-addNewCardButton.addEventListener("click", () => {
-  addCardFormPopup.open();
+api
+  .getInitialCards()
+  .then((res) => {
+    console.log(res);
+    cardSection.renderItems(res);
+  })
+
+  .catch((err) => alert(err));
+
+api
+  .getUserInfo()
+  .then((res) => {
+    userInfo.setUserInfo(res);
+    userInfo.updateProfileImage(res);
+  })
+  .catch((err) => alert(err));
+
+//Avatar
+const profileImageForm = document.querySelector("#edit-avatar-form");
```

Comment:
**COULD BE IMPROVED**

We can get any form using their **unique** attribute `name` or `id` from `document.forms`

```js

const profileForm = document.forms["profile-form"];
const cardForm = document.forms["card-form"];

```

You need to add `name` or `id` first in `index.html` to achieve that.

More info here https://developer.mozilla.org/en-US/docs/Web/API/Document/forms